source/d1min.cpp

```diff
@@ -5,61 +5,84 @@
 #include <QFileInfo>
 #include <QPainter>
 
-bool D1Min::load(QString filePath, quint16 subtileCount, std::map<unsigned, D1CEL_FRAME_TYPE> &celFrameTypes, const OpenAsParam &params)
+bool D1Min::load(QString filePath, int subtileCount, std::map<unsigned, D1CEL_FRAME_TYPE> &celFrameTypes, const OpenAsParam &params)
 {
-    // Opening MIN file with a QBuffer to load it in RAM
-    if (!QFile::exists(filePath))
-        return false;
-
-    QFile file = QFile(filePath);
-
-    if (!file.open(QIODevice::ReadOnly))
-        return false;
-
-    if (file.size() == 0)
-        return false;
+    // prepare file data source
+    QFile file;
+    // done by the caller
+    // if (!params.minFilePath.isEmpty()) {
+    //    filePath = params.minFilePath;
+    // }
+    if (!filePath.isEmpty()) {
+        file.setFileName(filePath);
+        if (!file.open(QIODevice::ReadOnly)) {
+            return false;
+        }
+    }
 
     QByteArray fileData = file.readAll();
     QBuffer fileBuffer(&fileData);
 
-    if (!fileBuffer.open(QIODevice::ReadOnly))
+    if (!fileBuffer.open(QIODevice::ReadOnly)) {
         return false;
-
-    // Read MIN binary data
-    QDataStream in(&fileBuffer);
-    in.setByteOrder(QDataStream::LittleEndian);
-
-    this->subtileWidth = params.minWidth;
-    if (this->subtileWidth == 0) {
-        this->subtileWidth = 2;
     }
-    this->subtileHeight = params.minHeight;
-    if (this->subtileHeight == 0) {
-        this->subtileHeight = file.size() / (subtileCount * this->subtileWidth * 2);
+
+    // calculate subtileWidth/Height
+    auto fileSize = file.size();
+    int width = params.minWidth;
+    if (width == 0) {
+        width = 2;
     }
-    if (file.size() != subtileCount * this->subtileWidth * this->subtileHeight * 2) {
-        qDebug() << "The size of sol-file does not align with min-file";
-        subtileCount = file.size() / (this->subtileWidth * this->subtileHeight * 2);
+    int height = params.minHeight;
+    if (height == 0) {
+        if (fileSize == 0 || subtileCount == 0) {
+            height = 5;
+        } else {
+            // guess subtileHeight based on the data
+            height = fileSize / (subtileCount * width * 2);
+        }
     }
 
     // File size check
-    if (file.size() % this->subtileHeight != 0)
+    int subtileNumberOfCelFrames = width * height;
+    if ((fileSize % (subtileNumberOfCelFrames * 2)) != 0) {
+        qDebug() << "Invalid min-file.";
```

Comment:
the min-file may be valid, it could be that we where given the wrong width/height

Reply:
yeah, but if you give the wrong width/height then the min-file is invalid to your use-case.

Reply:
```suggestion
        qDebug() << "Sub-tile width/height does not align with min-file.";
```